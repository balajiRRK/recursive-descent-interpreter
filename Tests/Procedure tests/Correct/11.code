procedure test is
	integer global;
	object divis;
	procedure func1 (object a, depth) is
		print(depth);
		
		divis['dividend'] = depth;
		integer i;
		for (i = depth - 1; 0 < i; i - 1) do
			divis['divisor'] = i;
			begin isDivis(divis);
			if divis == 1 then
				print(i);
			end
		end

		divis['dividend'] = a;
		divis['divisor'] = depth + 1;
		begin isDivis(divis);
		if divis == 1 then
			integer oldValue;
			oldValue = a;
			a = new object('0', oldValue + depth);
		end

		if not depth == 0 then
			object depth2;
			depth2 = new object('0', depth - 1);
			begin func1(a, depth2);
		else
			print(1000000);
		end

		a = a + 1;
		print(a);
	end

	procedure isDivis (object input) is
		if input['dividend'] == (input['dividend'] / input['divisor']) * input['divisor'] then
			input['return'] = 1;
		else
			input['return'] = 0;
		end
	end
begin
	divis = new object('return', 0);

	object depth;
	depth = new object('0', 16);
	object a;
	a = new object('0', 11);

	begin func1(a, depth);
end
