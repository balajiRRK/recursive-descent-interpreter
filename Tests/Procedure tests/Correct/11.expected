16
8
4
2
1
15
5
3
1
14
7
2
1
13
1
12
6
4
3
2
1
11
1
10
5
2
1
9
3
1
8
4
2
1
7
1
6
3
2
1
5
1
4
2
1
3
1
2
1
1
0
1000000
30
30
31
28
29
30
31
22
23
24
25
12
13
14
15
16
17
